lib/flowViewport/selection/fvpSelection.h

```diff
@@ -83,6 +87,20 @@ class Selection
     PXR_NS::HdDataSourceBaseHandle
     GetVectorDataSource(const PXR_NS::SdfPath& primPath) const;
 
+    //Get the wireframe color of a primitive for selection highlighting, 
+    // this checks if the prim is selected or not and if it is selected, 
+    // it returns the lead or active color depending on the lead of the selection
+    FVP_API
+    PXR_NS::GfVec4f GetWireframeColor(const PXR_NS::SdfPath& primPath)const;
+
+    //Ignore changes, this is used to keep the selection when we want to switch between display modes
+    FVP_API
+    void IgnoreChanges(bool val){_ignoreChanges = val;}
```

Comment:
That was the case, but let me check again since I did some modifications, maybe it's no longer needed.

Reply:
Here is the reason why I need this :
When updating the internal scene indices chain (not using any custom filtering scene index from the flow viewport API) after the merging scene index. We need to remove from the renderIndex the last scene index of the chain. In cascade, this produces a callstack of :
 	flowViewport.dll!Fvp_v0::Selection::RemoveHierarchy(const pxrInternal_v0_23__pxrReserved__::SdfPath & primPath) Line 145	C++
>	flowViewport.dll!Fvp_v0::SelectionSceneIndex::_PrimsRemoved(const pxrInternal_v0_23__pxrReserved__::HdSceneIndexBase & sender, const pxrInternal_v0_23__pxrReserved__::TfSmallVector<pxrInternal_v0_23__pxrReserved__::HdSceneIndexObserver::RemovedPrimEntry,16> & entries) Line 344	C++
 	usd_hd.dll!pxrInternal_v0_23__pxrReserved__::HdSceneIndexBase::_SendPrimsRemoved(const pxrInternal_v0_23__pxrReserved__::TfSmallVector<pxrInternal_v0_23__pxrReserved__::HdSceneIndexObserver::RemovedPrimEntry,16> & entries) Line 144	C++
 	usd_hd.dll!pxrInternal_v0_23__pxrReserved__::HdMergingSceneIndex::RemoveInputScene(const pxrInternal_v0_23__pxrReserved__::TfRefPtr<pxrInternal_v0_23__pxrReserved__::HdSceneIndexBase> & sceneIndex) Line 188	C++
 	flowViewport.dll!Fvp_v0::RenderIndexProxy::RemoveSceneIndex(const pxrInternal_v0_23__pxrReserved__::TfRefPtr<pxrInternal_v0_23__pxrReserved__::HdSceneIndexBase> & inputScene) Line 119	C++

So RemoveHierarchy is called and we lose the selection.

Reply:
I'd like to discuss this one with you.  It seems like we want to reconfigure the scene index tree, but leave scene index prims alone.  Multiple scene indices may have this requirement.  I have an intuition that we could create an insulating scene index filter that would simply stop propagating prims removed messages while we reconfigure the pipeline.  This would provide a service for multiple filtering scene indices.

Reply:
Let's discuss about that